File: src/main/java/com/mysawit/identity/MysawitIdentityServiceApplication.java
package com.mysawit.identity;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

@SpringBootApplication
public class MysawitIdentityServiceApplication {
    public static void main(String[] args) {
        ConfigurableApplicationContext context = SpringApplication.run(MysawitIdentityServiceApplication.class, args);
        if (context.getEnvironment().getProperty("app.test.close-context", Boolean.class, false)) {
            context.close();
        }
    }
}

File: src/main/kotlin/com/mysawit/identity/MysawitIdentityServiceApplication.kt
package com.mysawit.identity

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class MysawitIdentityServiceApplication

fun main(args: Array<String>) {
    val context = runApplication<MysawitIdentityServiceApplication>(*args)
    if (context.environment.getProperty("app.test.close-context", Boolean::class.java, false)) {
        context.close()
    }
}
